src/cml.js

```diff
@@ -303,6 +299,7 @@ class CML {
         }
       }
 
+      await exec(`git fetch`);
```

Comment:
Not sure if it's needed for your use case, it was for my tests.

Reply:
probably needed due to `actions/checkout` doing a shallow clone by default. Seems a little heavy-handed to undo that optimisation unconditionally but probably fine for now.

Reply:
> _probably needed due to actions/checkout doing a shallow clone by default._

Exactly that: `fetch-depth: 0` from `actions/checkout@v2` would also _(cough)_ solve it.

> _Seems a little heavy-handed to undo that optimisation unconditionally but probably fine for now._

It _is_ a bit heavy-handed. Do we have a more elegant option?

Reply:
Perhaps we could specify only the `<refspec>` we need?

Reply:
> Do we have a more elegant option?

Well the correct way would be a new option; something like:

```js
if (!opts.fetch) await exec('git fetch');
```

Reply:
> Perhaps we could specify only the `<refspec>` we need?

sure, even better (if possible)

Reply:
```suggestion
      await exec(`git fetch origin ${sha}`);
```

Reply:
It seems to work as expected